rasa/cli/x.py

```diff
@@ -291,16 +304,76 @@ def rasa_x(args: argparse.Namespace):
         run_locally(args)
 
 
-def run_in_production(args: argparse.Namespace):
-    from rasa.cli.utils import print_success
-    from rasa.core.utils import AvailableEndpoints
+def _pull_runtime_config_from_server(
+    config_endpoint: Optional[Text],
+    attempts: int = 30,
+    wait_time_between_pulls: Union[int, float] = 0.5,
+    keys: Tuple[Text, ...] = ("endpoints", "credentials"),
+) -> Tuple[Text, ...]:
+    """Pull runtime config from `config_endpoint`.
 
-    print_success("Starting Rasa X in production mode... 🚀")
+    Returns a path to yaml dumps, each containing the contents of one of `keys`.
+    """
+
+    while attempts > 0:
+        try:
+            response = requests.get(config_endpoint)
+            if response.status_code == 200:
+                rjs = response.json()
+                return tuple(_dump_dict_to_yaml(rjs, k) for k in keys)
+            else:
+                logger.debug(
+                    "Failed to get a proper response from remote "
+                    "server. Status Code: {}. Response: {}"
+                    "".format(response.status_code, response.text)
+                )
+        except requests.exceptions.ConnectionError as e:
+            logger.debug("Failed to connect to server. Retrying. {}".format(e))
+        time.sleep(wait_time_between_pulls)
+        attempts -= 1
+
+    print_error_and_exit(
+        "Could not fetch runtime config from server at '{}'. "
+        "Exiting.".format(config_endpoint)
+    )
+
+
+def _dump_dict_to_yaml(data: Dict, key: Text) -> Optional[Text]:
```

Comment:
can we put that in `utils.io` ?

Reply:
and test it please

Reply:
I'm not sure this should be a utils method. the method inspects the keys and exits if they're not present. Decoupling the inspection and dumping would make it too complex imo, and leave a lot of code in `x.py` anyway. what do you think?

Reply:
mhm, or raise an exception and then the caller can decide what to do with it? But you can leave it in -  I guess we will approach the point where we have to make `x` a package soon